db/merge_helper.cc

```diff
@@ -288,7 +288,9 @@ Status MergeHelper::MergeUntil(InternalIterator* iter,
     // Attempt to use the user's associative merge function to
     // merge the stacked merge operands into a single operand.
     s = Status::MergeInProgress();
-    if (merge_context_.GetNumOperands() >= 2) {
+    if (merge_context_.GetNumOperands() >= 2 ||
+        (merge_context_.GetNumOperands() == 1 &&
+         user_merge_operator_->DoesAllowSingleMergeOperand())) {
```

Comment:
Instead of calling this virtual function every time we see a single merge operand (which can be a lot), can we call it when constructing MergeHelper and store it as a member variable?